ddtrace/internal/datadog/profiling/stack/echion/echion/threads.h

```diff
@@ -221,7 +221,7 @@ ThreadInfo::unwind_tasks(PyThreadState* tstate)
     std::unordered_set<PyObject*> parent_tasks;
     std::unordered_map<PyObject*, TaskInfo::Ref> waitee_map; // Indexed by task origin
     std::unordered_map<PyObject*, TaskInfo::Ref> origin_map; // Indexed by task origin
-    static std::unordered_set<PyObject*> previous_task_objects;
+    static std::unordered_map<PyObject*, StringTable::Key> previous_task_objects;
```

Comment:
What's the purpose of changing this to a hash-table? It's not being used any different in this PR, far as I can tell. Do you have plans for it in the future?